openeo/extra/job_management.py

```diff
@@ -886,3 +937,124 @@ def create_job_db(path: Union[str, Path], df: pd.DataFrame, *, on_exists: str =
     else:
         raise NotImplementedError(f"Initialization of {type(job_db)} is not supported.")
     return job_db
+
+
+class UDPJobFactory:
+    """
+    Batch job factory based on a parameterized process definition
+    (e.g a user-defined process (UDP) or a remote process definition),
+    to be used together with :py:class:`MultiBackendJobManager`.
+    """
+
+    def __init__(
+        self,
+        process_id: str,
```

Comment:
Also allow process_id to be None, e.g. to define process with just the namespace